leshan-server-core/src/main/java/org/eclipse/leshan/server/impl/LwM2mRequestSenderImpl.java

```diff
@@ -94,4 +94,13 @@ public void stop() {
             ((Stoppable) defaultRequestSender).stop();
         }
     }
+
+    @Override
+    public void cancelPendingRequests(Client client) {
+        if (client.usesQueueMode()) {
+            queuedRequestSender.cancelPendingRequests(client);
+        } else {
+            defaultRequestSender.cancelPendingRequests(client);
+        }
+    }
```

Comment:
You must cancel request on both 2 senders because, for the same registration you could have pending request in queuemode and not queuemode (as binding mode can change)
(I agree this Far-fetched but I see no problem to just call `cancelPendingRequests` on both sender)